plugins/modules/ndo_service_device_cluster.py

```diff
@@ -0,0 +1,641 @@
+        min_threshold: 10
+        max_threshold: 100
+        threshold_down_action: permit
+      - name: interface3
+        type: bd
+        interface:
+          name: ansible_test_bd
+          template: ansible_template
+          schema_id: '{{ ansible_test_bd.current.vrfRef.schemaId }}'
+        qos_policy:
+          name: ansible_custom_qos_policy
+          template_id: '{{ ansible_test_policy.current.templateId }}'
+        anycast: true
+        rewrite_source_mac: false
+        config_static_mac: false
+        is_backup_redirect_ip: true
+        load_balance_hashing: destinationIP
+        pod_aware_redirection: false
+        resilient_hashing: true
+        tag_based_sorting: false
+        min_threshold: 10
+        max_threshold: 100
+        threshold_down_action: deny
+    state: present
+
+- name: Query the service device cluster using name
+  cisco.mso.ndo_service_device_cluster:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_device_template
+    name: device1
+    state: query
+  register: query_device
+
+- name: Query the second service device cluster using UUID
+  cisco.mso.ndo_service_device_cluster:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_device_template
+    uuid: '{{ add_device1.current.uuid }}'
+    state: query
+  register: query_device_uuid
+
+- name: Query all the device clusters
+  cisco.mso.ndo_service_device_cluster:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_device_template
+    state: query
+  register: query_all
+
+- name: Remove the service device cluster
+  cisco.mso.ndo_service_device_cluster:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_device_template
+    name: device1
+    state: absent
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+    MSOTemplateCache,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.utils import (
+    append_update_ops_data,
+    check_if_all_elements_are_none,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.schema import (
+    MSOSchemaCache,
+)
+import copy
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str"),
+        template_id=dict(type="str"),
+        name=dict(type="str"),
+        uuid=dict(type="str"),
+        description=dict(type="str"),
+        device_mode=dict(type="str", choices=["layer1", "layer2", "layer3"]),
+        device_type=dict(type="str", choices=["firewall", "loadBalancer", "other"]),
+        interface_properties=dict(
+            type="list",
+            elements="dict",
+            options=dict(
```

Comment:
should we define required_one_of and mutually_exclusive for interface and interface uuid?

Reply:
same applies for ipsla, qos.